common.go

```diff
@@ -70,3 +90,175 @@ func GatherMetadata(wh *workloads.WorkloadHelper, alerting bool) error {
+	for _, storageClass := range storageClasses.Items {
+		if val, ok := storageClass.Annotations[defaultStorageClassAnnotation]; ok && val == "true" {
+			defaultStorageClassName = storageClass.GetName()
+		}
+		if val, ok := storageClass.Annotations[defaultVirtStorageClassAnnotation]; ok && val == "true" {
+			defaultVirtStorageClassName = storageClass.GetName()
+		}
+	}
+	if preferVirt {
+		storageClassName = defaultVirtStorageClassName
+	}
+	if storageClassName == "" {
+		storageClassName = defaultStorageClassName
+	}
+	if storageClassName == "" {
+		return "", fmt.Errorf("storageClassName was not provided with no default StorageClass")
+	}
+	return storageClassName, nil
+}
+
+func storageClassSupportsVolumeExpansion(storageClassName string) bool {
+	kubeClientProvider := config.NewKubeClientProvider("", "")
+	clientSet, _ := kubeClientProvider.ClientSet(0, 0)
+
+	storageClass, err := clientSet.StorageV1().StorageClasses().Get(context.Background(), storageClassName, v1.GetOptions{})
+	if err != nil {
+		log.Errorf("Failed to get storageClass resource - %v", err)
+		return false
+	}
+
+	return storageClass.AllowVolumeExpansion != nil && *storageClass.AllowVolumeExpansion
+}
+
+func getStorageClassProvisioner(storageClassName string) (string, error) {
+	kubeClientProvider := config.NewKubeClientProvider("", "")
+	clientSet, _ := kubeClientProvider.ClientSet(0, 0)
+
+	storageClass, err := clientSet.StorageV1().StorageClasses().Get(context.Background(), storageClassName, v1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	return storageClass.Provisioner, nil
+}
+
+func getDynamicClient() *dynamic.DynamicClient {
+	_, restConfig := config.NewKubeClientProvider("", "").ClientSet(0, 0)
+	return dynamic.NewForConfigOrDie(restConfig)
+}
+
+func getVolumeSnapshotClassNameForProvisioner(provisioner string) (string, error) {
+	volumeSnapshotClassGVR := schema.GroupVersionResource{Group: "snapshot.storage.k8s.io", Version: "v1", Resource: "volumesnapshotclasses"}
+	client := getDynamicClient()
+	itemList, err := client.Resource(volumeSnapshotClassGVR).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed listing %s: %v", volumeSnapshotClassGVR.Resource, err)
+	}
+	var snapshotClassName string
+	for _, item := range itemList.Items {
+		driver, found, err := unstructured.NestedString(item.Object, "driver")
+		if err != nil || !found {
+			return "", fmt.Errorf("failed to get driver for %s/%s - %v", volumeSnapshotClassGVR.Resource, item.GetName(), err)
+		}
+		if driver == provisioner {
+			snapshotClassName = item.GetName()
+			break
+		}
+	}
+	return snapshotClassName, nil
+}
+
+func getVolumeSnapshotClassNameForStorageClass(storageClassName string) (string, error) {
+	provisioner, err := getStorageClassProvisioner(storageClassName)
+	if err != nil {
+		return "", err
+	}
+	return getVolumeSnapshotClassNameForProvisioner(provisioner)
+}
+
+func virtctlExists() bool {
```

Comment:
Some of the functions in this file seem to be commonly used across all the virtualization portifolio. Can we move all of them to go-commons please?